includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2184,7 +2187,7 @@ public function cancel_authorization( $order ) {
 		}
 
 		if ( 'canceled' === $status ) {
-			$this->order_service->mark_payment_capture_cancelled( $order, $intent->get_id(), $status, $intent->get_charge_id() );
+			$this->order_service->mark_payment_capture_cancelled( $order, $intent->get_id(), $status, $intent->get_charge()->get_id() );
```

Comment:
The same applies here. If `get_charge` returns `null`, we'll end up calling a function on `null`. In case it's possible that we call this for an intent without a charge, we should probably check whether the intent has a charge before getting the charge ID to prevent that.